subprojects/GOPATH/src/github.com/microsoft/hivedscheduler/pkg/api/types.go

```diff
@@ -96,15 +96,20 @@ type PodBindInfo struct {
 	// The node to bind
 	Node string `yaml:"node"`
 	// The GPUs to bind
-	GpuIsolation          []int32  `yaml:"gpuIsolation"`
-	CellChain             string   `yaml:"cellChain"`
-	PodCellLevel          int32    `yaml:"podCellLevel"`
-	PodCellGpuPlacement   Range    `yaml:"podCellGpuRange"`
-	GroupCellLevel        int32    `yaml:"groupCellLevel"`
-	GroupCellNodes        []string `yaml:"groupCellNodes"`
-	GroupCellGpuPlacement Range    `yaml:"groupCellGpuRange"`
-	VirtualCellLevel      int32    `yaml:"virtualCellLevel"`
-	VirtualCellIndex      int32    `yaml:"virtualCellIndex"`
+	GpuIsolation      []int32                   `yaml:"gpuIsolation"`
+	CellChain         string                    `yaml:"cellChain"`
+	GroupScheduleInfo []GroupMemberScheduleInfo `yaml:"groupScheduleInfo"`
+}
+
+type GroupMemberScheduleInfo struct {
+	GpuNumber     int32              `yaml:"gpuNumber"`
+	PodPlacements []PodPlacementInfo `yaml:"podPlacements"`
+}
+
+type PodPlacementInfo struct {
+	PhysicalNode          string  `yaml:"physicalNodes"`
+	PhysicalGpuIndices    []int32 `yaml:"physicalGpuIndices"`
+	VirtualGpuCellIndices []int32 `yaml:"virtualGpuCellIndices"`
```

Comment:
>VirtualGpuCellIndices [](start = 1, length = 21)

Why not VirtualCellIndices?
And then:
type PodPlacementInfo struct {
 Node          string
 GpuIndices    []int32
 CellIndices []int32
}

Reply:
Renamed. But the GPU indices is for physical, cell indices is for virtual. I'm afraid your simplification would lead to confusion

---
In reply to: [332496961](https://github.com/microsoft/pai/pull/3684#discussion_r332496961) [](ancestors = 332496961)